### PR TITLE
feat(robot-server): http basic pipetting- load labware

### LINKED ISSUE #6591: feat(robot-server): handle load labware command in live protocol session
## overview

In order to run a basic pipetting protocol over HTTP, a user must be able to issue a load labware command over HTTP to add a specific piece of labware to the deck.

## acceptance criteria

- [ ] User can issue a `loadLabware` command
    - Payload already defined as part of #6560, includes loadName, namespace, version, and slot
- [ ] After the command has been executed, the session response should include a new labware resource
    - Labware resource should have a unique ID for future commands to reference
    - Labware resource should include properties from the command
- [ ] After the command has been executed, the labware's definition and calibration data should be loaded into the protocol executor's state store
    - Loading the definition into state will provide the easiest path to common machinery between "load labware" and "load labware from definition" commands

### out of scope but on our minds

- Avoiding labware slot collisions
- Loading labware on modules
- Loading labware from definitions
- Returning full definitions of loaded labware from the HTTP API

## implementation notes

- Command implementation for the command will probably look something like this:
    - Side-effect: generate unique identifier for the labware resource
    - Side-effect: load labware definition from file-system (via established mechanisms)
    - Side-effect: load labware calibration data from file-system (via established mechanisms)
    - Command result: labware definition, labware calibration data
- We'll probably want to create some sort of labware state subtree that includes:
    - Entry for labware definition
    - Entry for labware calibration data
        - Should be separate from definition so it can be independently updated
    - Identifier linkage to definition entry, calibration data entry, and deck location

### PR DESCRIPTION
**Pull request by sanni-t**
# Overview

Live protocol session continued! This PR adds the ability to load labware.
Closes #6591 

# Changelog

- added load_labware handler
- added LoadLabwareResponse model
- added labware state handler to store
- added unit tests & tavern integration tests

# Risk assessment

Low. Adds a 'result' router field to sessions command executor. Everything else is still isolated.

### CODE REVIEW

**Review by amitlissack — APPROVED** [review]
🥇

**Review by sfoster1 — COMMENTED** [review]
The stuff that's implemented looks good, but I have two kind of larger points -
1. The choice we made with the existing protocol stack to have clients send custom labware definitions - to structurally have custom and maybe all labware definitions live on the client and be sent on-demand to the robot - isn't actually driven by the problems with the existing protocol stack, it's because of larger problems with the distributed nature of runapp + server. I don't think it'll add much code to have a load labware request where the client can upload a definition, and I think it's a good thing to have in from the start. 
2. This PR loads labware objects into an internal store but not into a deck object, which is a minor thing but it's worth noting that the deck is where some of the logic around like highest-point tracking lives so you'll probably want to do that before implementing motion control stuff.

File: robot-server/robot_server/service/session/session_types/live_protocol/state_store.py [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
+    definition: LabwareDefinition
+    calibration: OffsetVector
+    deckLocation: int
+
+
 class StateStore:
     def __init__(self):
         self._commands: List[Command] = []
         self._command_results_map: Dict[str, CommandResult] = dict()
+        self._handler_map: Dict[
```
Comment (mcous) [inline robot-server/robot_server/service/session/session_types/live_protocol/state_store.py:23]: [No change requested] As discussed over the call with everyone: as this grows, I suspect a system by which a given substate knows which commands to handle (as opposed to this, where the parent needs to know how to route commands) will be more flexible and scalable

File: robot-server/tests/service/session/session_types/live_protocol/test_state_store.py [hunk 2]
```diff
@@ -23,23 +25,65 @@ def test_handle_command_request():
...
+            command.meta.identifier) == command_result
+
+
+@pytest.mark.parametrize(
+    argnames="command_name, handler",
+    argvalues=[[models.EquipmentCommand.load_labware, "handle_load_labware"]]
+)
+def test_command_result_state_handler(command_name, handler):
+
+    with patch.object(StateStore, handler) as handler_mock:
```
Comment (mcous) [inline robot-server/tests/service/session/session_types/live_protocol/test_state_store.py:57]: Following up on our discussion about why I had an impulse to remove this test, I wanted to recommend this [article about why partial mocks should be avoided](https://github.com/testdouble/contributing-tests/wiki/Partial-Mock) by the author of the excellent [testdouble.js](https://github.com/testdouble/testdouble.js) mocking library.



(Honestly the [entire wiki](https://github.com/testdouble/contributing-tests/wiki/Test-Driven-Development) the article lives in is a great read if you have some time)

**Review by mcous — APPROVED** [review]
🤖

**Review by sfoster1 — APPROVED** [review]
Does look good!

**Comment by mcous** [comment]
Thanks for the thoughts, @sfoster1!

> The choice we made with the existing protocol stack to have clients send custom labware definitions - to structurally have custom and maybe all labware definitions live on the client and be sent on-demand to the robot - isn't actually driven by the problems with the existing protocol stack, it's because of larger problems with the distributed nature of runapp + server. I don't think it'll add much code to have a load labware request where the client can upload a definition, and I think it's a good thing to have in from the start.

This is a good point. We have talked about and planned a `loadLabwareFromDefinition` command (or similar), and it's looking like the first requirement for such a command will be driven by the JSON protocol executor. Since that's basically next, it'll be included imminently. The need to load labware from definitions directly is why we chose to insert the definition into state, so loading by load name vs definition basically become the same thing immediately after the command is "handled", because then the system is just dealing with "I have this labware and its definition is in state".

> This PR loads labware objects into an internal store but not into a deck object, which is a minor thing but it's worth noting that the deck is where some of the logic around like highest-point tracking lives so you'll probably want to do that before implementing motion control stuff.

Thanks, we'll keep an eye on that. We're trying (I hope in an obvious way) to use what's already there more than anything else. `Deck` has the concern of being a thing that does logical geometry calculations but _also_ needs to act as a public interface for the Python Protocol API, so we may end up doing some extraction (or temporary duplication) to avoid conflicts

**Pull request merged**